tls/s2n_prf.c

```diff
@@ -292,9 +300,17 @@ int s2n_prf_server_finished(struct s2n_connection *conn)
     master_secret.data = conn->pending.master_secret;
     master_secret.size = sizeof(conn->pending.master_secret);
     if (conn->actual_protocol_version == S2N_TLS12) {
-        GUARD(s2n_hash_digest(&conn->handshake.server_sha256, sha_digest, SHA256_DIGEST_LENGTH));
-        sha.data = sha_digest;
-        sha.size = SHA256_DIGEST_LENGTH;
+        if (conn->pending.cipher_suite->tls12_prf_alg == S2N_HMAC_SHA256) {
+            GUARD(s2n_hash_digest(&conn->handshake.server_sha256, sha_digest, SHA256_DIGEST_LENGTH));
+            sha.data = sha_digest;
```

Comment:
A switch statement would be less code, and either way you can pull out the sha.data = line, since it's the same in both non-terminal branches.